### ISSUE TITLE
model.generate() fails

### ISSUE THREAD

**Issue by noy-sternlicht** [fleonce/ITER#3]
Hey :) Running the example code snippet from Hugging Face:

```
    model = ITERForRelationExtraction.from_pretrained("fleonce/iter-scierc-scideberta-full")
    tokenizer = model.tokenizer

    encodings = tokenizer(
        "An art exhibit at the Hakawati Theatre in Arab east Jerusalem was a series of portraits of Palestinians killed in the rebellion .",
        return_tensors="pt"
    )

    generation_output = model.generate(
        encodings["input_ids"],
        attention_mask=encodings["attention_mask"],
    )

    # entities
    print(generation_output.entities)

    # relations between entities
    print(generation_output.links)
```

fails with the following assertion:

```
  File "/test.py", line 13, in main
    generation_output = model.generate(
                        ^^^^^^^^^^^^^^^
  File "/myenv/lib/python3.11/site-packages/iter/modeling_iter2.py", line 254, in generate
    generation_output = self._generate_forward(
                        ^^^^^^^^^^^^^^^^^^^^^^^
  File "/myenv/lib/python3.11/site-packages/iter/modeling_iter2.py", line 801, in _generate_forward
    outer = proj(is_right_hidden, False, False)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/myenv/lib/python3.11/site-packages/iter/modeling_iter2.py", line 798, in proj
    return unsqueeze_and_expand_dims(inp, unsqueeze, *expands)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/myenv/lib/python3.11/site-packages/torchcheck/ops/unsqueeze.py", line 40, in unsqueeze_and_expand_dims
    return expand_dims(self, dims_and_sizes)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/myenv/lib/python3.11/site-packages/torchcheck/ops/expand.py", line 64, in expand_dims
    assert not isinstance(dims_and_sizes[0], int)
AssertionError
```

Is it a dependency issue? I noticed you don't have a requirement file, and I'm unsure which versions I should use to run your code.

**Comment by fleonce** [comment]
Looks like a dependency issue, I will investigate and get back to you shortly.

In the meantime, you can check whether the pip install installed the correct versions by referring to the `pyproject.toml` where the dependencies are listed.

I think supplying a requirements.txt would probably be a good idea, but the `pyproject.toml` should be sufficient

**Comment by fleonce** [comment]
Unfortunately it was not a dependency problem, but due to the example being from another dataset (CoNLL04), generation returns no entities and no entities resulted in an exception during testing for relationships (between 0 entities)

This should be fixed now.

To reinstall, you can use something along those lines:
`pip install git+https://github.com/fleonce/ITER --force-reinstall`

**Comment by noy-sternlicht** [comment]
Hey, thanks for the quick response! 
I think the issue is not resolved... I keep getting a similar error (after reinstalling the package) for sentences that should include entities. For example:  "A new approach for Interactive Machine Translation, where the author interacts during the creation or modification of the document, is proposed." which is taken from the SciERC dev dataset.

**Comment by fleonce** [comment]
Just to keep you informed, I will push a fix in the next hours. I moved the code to a new codebase and also rewrote some parts for better readability, and apparently generation for nested named entity datasets is currently broken.

**Comment by fleonce** [comment]
@noy-sternlicht I believe this issue should be resolved now as well. Thank you for pointing this out!

When I run the following command locally (cpu or gpu), I get the exact score reported for the model on huggingface:
```shell
python3 evaluate.py -m fleonce/iter-scierc-scideberta-full
```

which just performs one evaluation run on the test set of scideberta

Feel free to reach out if you encounter any additional issues and/or need further help. Thanks again!

**Comment by noy-sternlicht** [comment]
Of course :) Thanks, it works now!
